sktime/forecasting/base/_fh.py

```diff
@@ -273,23 +273,23 @@ def to_relative(self, cutoff=None):
             absolute = self.to_pandas()
             _check_cutoff(cutoff, absolute)
 
-            if isinstance(absolute, pd.DatetimeIndex):
-                # We cannot use the freq from the the ForecastingHorizon itself (or its
-                # wrapped pd.DatetimeIndex) because it may be none for non-regular
-                # indices, so instead we use the freq of cutoff.
-                freq = _get_freq(cutoff)
+            # We cannot use the freq from the ForecastingHorizon itself (or its
+            # wrapped pd.DatetimeIndex) because it may be none for non-regular
+            # indices, so instead we use the freq of cutoff.
+            freq = _get_freq(cutoff)
 
+            if isinstance(absolute, pd.DatetimeIndex):
                 # coerce to pd.Period for reliable arithmetics and computations of
                 # time deltas
                 absolute = _coerce_to_period(absolute, freq)
                 cutoff = _coerce_to_period(cutoff, freq)
 
             # Compute relative values
-            relative = absolute - cutoff
+            relative = pd.Index([date - cutoff for date in absolute])
```

Comment:
can we avoid this back/forth conversion?

Reply:
this specific line is exactly the hack we need to circumvent the bug in pandas with difference between `PeriodIndex` and `Period` objects. converting the list to index is necessary as further down the road it is only index, but not a list is expected